src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elements;
+
+    public ArrayList(int capacity) {
+        this.elements = (T[])new Object[capacity];
```

Comment:
Avoid using Object[] for creating a generic array. Consider using a different approach that doesn't involve casting to T[].